### PR TITLE
SAAS-175 Add tests for auth API

### PR DESCRIPTION
**Pull request by artemgavrilov**
- https://jira.percona.com/browse/SAAS-175

- [x] https://github.com/percona/pmm-managed/pull/420
- [x] https://github.com/percona/pmm/pull/553

Feature build: Percona-Lab/pmm-submodules#967

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -10,4 +10,4 @@ required = [
 
 [[constraint]]
   name = "github.com/percona/pmm"
-  branch = "PMM-2.0"
+  branch = "SAAS-175-auth-api"
```
Comment (artemgavrilov) [inline Gopkg.toml:13]: TODO

File: server/percona_auth_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/percona/pmm/api/serverpb/json/client/server"
+	"github.com/stretchr/testify/require"
+
+	pmmapitests "github.com/Percona-Lab/pmm-api-tests"
+)
+
+func TestPerconaAuth(t *testing.T) {
+	client := serverClient.Default.Server
+	login := gofakeit.Email()
+	password := "Password12345"
```
Comment (palash25) [inline server/percona_auth_test.go:18]: same as this one https://github.com/percona/pmm-managed/pull/420#discussion_r456396635
consider using gofakeit for password generation

Comment (artemgavrilov) [reply server/percona_auth_test.go]: Done

File: server/platform_auth_test.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+	"github.com/brianvoe/gofakeit"
+	serverClient "github.com/percona/pmm/api/serverpb/json/client"
+	"github.com/percona/pmm/api/serverpb/json/client/server"
+	"github.com/stretchr/testify/require"
+
+	pmmapitests "github.com/Percona-Lab/pmm-api-tests"
+)
+
+// Tests in that file cover Percona Platform authentication.
```
Comment (palash25) [inline server/platform_auth_test.go:15]: `that` -> `this`, rest looks fine to me

Comment (artemgavrilov) [reply server/platform_auth_test.go]: Fixed

File: server/auth_test.go [hunk 4]
```diff
@@ -239,40 +239,66 @@ func TestPermissionsForSTTChecksPage(t *testing.T) {
...
+			for _, user := range test.userCase {
+				user := user
+				t.Run(user.userType, func(t *testing.T) {
+					// make a BaseURL without authentication
+					u, err := url.Parse(pmmapitests.BaseURL.String())
+					require.NoError(t, err)
+					u.User = url.UserPassword(user.login, user.login)
+					u.Path = test.url
+
+					req, err := http.NewRequest(test.method, u.String(), nil)
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply server/auth_test.go:292]: Good bot

[the remark this answers is not in the document]
Comment (artemgavrilov) [reply server/auth_test.go:292]: Does it have any sense in tests? We have a global timeout for tests execution.

[the remark this answers is not in the document]
Comment (AlekSi) [reply server/auth_test.go:292]: Yeah, we need it for cleanup.

https://github.com/Percona-Lab/pmm-api-tests/blob/877d5376e60c4a66eac3d81e2027e546fd2f0180/init.go#L32-L33

[the remark this answers is not in the document]
Comment (artemgavrilov) [reply server/auth_test.go]: Done

File: server/platform_auth_test.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+// Tests in this file cover Percona Platform authentication.
+
+func TestPlatformAuth(t *testing.T) {
+	client := serverClient.Default.Server
+	login := gofakeit.Email()
+	password := gofakeit.Password(true, true, true, false, false, 14)
+
+	_, err := client.PlatformSignUp(&server.PlatformSignUpParams{
```
Comment (BupycHuk) [inline server/platform_auth_test.go:21]: Should we check result of request?

Comment (artemgavrilov) [reply server/platform_auth_test.go]: It's empty structure, there aren't any fields to check

**Pull request merged**